guava/src/com/google/common/base/Predicates.java

```diff
@@ -234,9 +234,11 @@ public static Predicate<Class<?>> subtypeOf(Class<?> clazz) {
    *
    * @throws IllegalArgumentException if the pattern is invalid
    * @since 3.0
+   * @deprecated Use {@link java.util.regex.Pattern#asPredicate}
    */
   @J2ktIncompatible
   @GwtIncompatible // Only used by other GWT-incompatible code.
+  @Deprecated
   public static Predicate<CharSequence> containsPattern(String pattern) {
     return new ContainsPatternFromStringPredicate(pattern);
   }
```

Comment:
CC @cushon: Error Prone `CompilationTestHelper` and `DiagnosticTestHelper` users are [nudged](https://github.com/search?q=repo%3Agoogle%2Ferror-prone%20expectErrorMessage&type=code) to using this API, and because `expectErrorMessage` expects a `com.google.common.base.Predicate` rather than a `java.util.function.Predicate`, the suggested alternative won't readily work. (The workaround is to use `s -> Pattern.compile(pattern).find(s)`, though that doesn't make for as nice an API. :thinking:)

Reply:
Good call, I will look at updating those APIs